nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -1008,7 +1009,8 @@ Nullness returnValueNullness(
       nullness = input.getRegularStore().valueOfMethodCall(node, state, NULLABLE, apContext);
     } else if (node == null
         || methodReturnsNonNull.test(node)
-        || !Nullness.hasNullableAnnotation((Symbol) node.getTarget().getMethod(), config)) {
+        || (!Nullness.hasNullableAnnotation((Symbol) node.getTarget().getMethod(), config)
+            && !genericReturnIsNullable(node))) {
```

Comment:
Our current implementation efforts are prioritizing checking typical _clients_ of generic classes, rather than generic classes themselves.  There are all kinds of cases involving generic type _variables_ that we are not checking right now.  So, for now, we will miss reporting errors like for your example, where there is a dereference of a variable whose type is a generic type variable (which can be either `@Nullable` or `@NonNull`).  We do not have any tests around these scenarios yet as this support is lower priority IMO.  Hopefully that's ok.